src/hotspot/share/gc/z/zNMT.cpp

```diff
@@ -28,92 +28,28 @@
 #include "gc/z/zVirtualMemory.hpp"
 #include "memory/allocation.hpp"
 #include "nmt/memTracker.hpp"
+#include "nmt/nmtMemoryFileTracker.hpp"
 #include "utilities/nativeCallStack.hpp"
 
-ZNMT::Reservation ZNMT::_reservations[ZMaxVirtualReservations] = {};
-size_t ZNMT::_num_reservations = 0;
+MemoryFileTracker::MemoryFile* ZNMT::_device = nullptr;
 
-size_t ZNMT::reservation_index(zoffset offset, size_t* offset_in_reservation) {
-  assert(_num_reservations > 0, "at least one reservation must exist");
-
-  size_t index = 0;
-  *offset_in_reservation = untype(offset);
-  for (; index < _num_reservations; ++index) {
-    const size_t reservation_size = _reservations[index]._size;
-    if (*offset_in_reservation < reservation_size) {
-      break;
-    }
-    *offset_in_reservation -= reservation_size;
-  }
-
-  assert(index != _num_reservations, "failed to find reservation index");
-  return index;
+void ZNMT::reserve(zaddress_unsafe start, size_t size) {
+  MemTracker::record_virtual_memory_reserve((address)start, size, CALLER_PC, mtJavaHeap);
```

Comment:
I'd prefer if we scale away the zaddress_unsafe type before casting to a pointer:
```suggestion
  MemTracker::record_virtual_memory_reserve((address)untype(start), size, CALLER_PC, mtJavaHeap);
```